dialogue-apache-hc5-client/src/test/java/com/palantir/dialogue/hc5/ApacheHttpClientChannelsTest.java

```diff
@@ -227,4 +258,20 @@ private long poolGaugeValue(TaggedMetricRegistry metrics, String clientName, Str
         assertThat(value).isInstanceOf(Long.class);
         return (long) value;
     }
+
+    @SuppressWarnings("JdkObsolete")
+    private long unknownHostCount(TaggedMetricRegistry metrics, String clientName) {
+        Set<Metric> matchingCounters = metrics.getMetrics().entrySet().stream()
+                .filter(entry -> entry.getKey().safeName().equals("dialogue.client.connection.resolution.error"))
+                .filter(entry -> clientName.equals(entry.getKey().safeTags().get("client-name")))
+                .map(Map.Entry::getValue)
+                .collect(Collectors.toSet());
+        if (matchingCounters.isEmpty()) {
+            return 0L;
+        }
+
+        Metric counter = Iterables.getOnlyElement(matchingCounters);
+        assertThat(counter).isInstanceOf(Counter.class);
+        return ((Counter) counter).getCount();
+    }
```

Comment:
We can simplify + inline
```suggestion
    private long unknownHostCount(TaggedMetricRegistry metrics, String clientName) {
        return  DialogueClientMetrics.of(metrics).connectionResolutionError(clientName);
    }
```

Reply:
Are gauges different? If no, should I also fix the other method (I stole this structure from)?

Reply:
Gauges are different, they're runnable tasks which return a value, so the equivalent method would attempt to register a new one.
We added some methods for gauges (likely after that test was written) which allow us to build the MetricName object. That may clean things up, but I think it's out of scope for this PR.

Reply:
Gauges are different! 😄